lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -125,7 +125,14 @@ def matches_arity?(actual, name)
 
           return true if expectation.empty?
 
-          signature = Support::MethodSignature.new(Support.method_handle_for(actual, name))
+          method_handle = Support.method_handle_for(actual, name)
+
+          signature =
+            if name == :new && method_handle.owner === ::Class && Class === actual
```

Comment:
Why use `::Class` in one spot and `Class` (no `::` in another)?

Reply:
Oversight, fixed